MRL_Mother/MRL_MotherModel/MRL_AI_Mother_Autonomous_Runtime_Baseline_v1/runtime/MRL_local_model_adapter_v1.py

```
@@ -0,0 +1,128 @@
+        )
+    return endpoint.rstrip("/")
+
+
+def _json_request(url: str, payload: dict[str, Any] | None, timeout: int) -> dict[str, Any]:
+    data = None if payload is None else json.dumps(payload).encode("utf-8")
+    request = urllib.request.Request(
+        url,
+        data=data,
+        headers={"Accept": "application/json", "Content-Type": "application/json"},
+        method="GET" if payload is None else "POST",
+    )
```

Ruff (0.16.2)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)